packages/db/postgres/functions.sql

```diff
@@ -251,17 +251,38 @@ BEGIN
 END
 $$;
 
-CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) RETURNS TEXT
-  LANGUAGE plpgsql
+CREATE TYPE used_storage AS (uploaded TEXT, pinned TEXT);
+
+CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) 
```

Comment:
what is the purpose of this table?  if an admin increases the user's storage limit, will entries in this table be out of date or invalid?

Reply:
@the-simian had another valid use case: if an admin _decreases_ the user's limit.  are these scenarios handled?

Reply:
This is an addition to the existing function `user_used_storage` which gets the current amount of storage used for a specified user, for display on the web3 users `account` page. The extra SQL statement gets the current amount of pinned storage used and returns that as well.

The amount of storage used is independent of the users storage limit. A related ticket #990 builds on this and checks the calculated storage amounts against the user's quota limits, to notify them when they are approaching their limit.

Reply:
i feel dumb.  not sure how i missed that this was a function, not a table.  thank you!